share/spice/upc/upc.js

```diff
@@ -0,0 +1,44 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_upc = function(api_result){
+
+        if (api_result.valid == 'false') {
+            return Spice.failed('upc');
+        }
+
+        Spice.add({
+            id: "upc",
+            name: "Universal Product Code (UPC)",
+            data: api_result,
+            meta: {
+                sourceName: "upcdatabase.org",
+                sourceUrl: 'http://upcdatabase.org/code/' + api_result.number
+            },
+            normalize: function(item) {
+                if (item.valid == 'false') {
+                    return Spice.failed('upc');
+                }
+
+                var alias = (item.alias != '') ? item.alias : item.itemname;
+                var avg_price = (item.avg_price != '') ? item.avg_price : "0";
+                var rate_up = (item.rate_up != '') ? item.rate_up : "0";
+                var rate_down = (item.rate_down != '') ? item.rate_down : "0";
+
+                return {
+                    alias: alias,
+                    avg_price: avg_price,
+                    rate_up: rate_up,
+                    rate_down: rate_down,
+                    title: item.itemname
```

Comment:
Or it could say, "It has an average ...". Looks like it's still valid just missing a title, I guess.